spec/podfile/dsl_spec.rb

```
@@ -679,6 +679,102 @@ module Pod
+          ensure_bundler!
+        end
+        restore_bundler_environment
+        CoreUI.warnings.should == "CocoaPods was invoked from Global Gemset.\nPlease re-run using: `bundle exec pod #{ARGV.join(' ')}`"
+      end
+
+      it 'Prints warning when a bundler env variable not found' do
+        stash_bundler_environment
+        ENV['BUNDLE_GEMFILE'] = nil
+        Podfile.new do
+          ensure_bundler!
+        end
+        restore_bundler_environment
+        CoreUI.warnings.should == "CocoaPods was invoked from Global Gemset.\nPlease re-run using: `bundle exec pod #{ARGV.join(' ')}`"
+      end
+
+      it 'Has no warning with Bundler environment' do
+        stash_bundler_environment
+        ENV['BUNDLE_GEMFILE'] = 'mock argument'
+        ENV['BUNDLE_BIN_PATH'] = 'another mock argument'
+        ENV['BUNDLER_VERSION'] = '1.17.1'
+        Podfile.new do
+          ensure_bundler!
+        end
+        restore_bundler_environment
+        CoreUI.warnings.should.be.empty
+      end
+
+      it 'Prints version warning when version argument fails match' do
+        stash_bundler_environment
+        ENV['BUNDLE_GEMFILE'] = 'mock argument'
+        ENV['BUNDLE_BIN_PATH'] = 'another mock argument'
+        ENV['BUNDLER_VERSION'] = '1.17.1'
+        Podfile.new do
+          ensure_bundler! '2.0.0'
+        end
+        restore_bundler_environment
+        CoreUI.warnings.should == 'The installed Bundler version: 1.17.1 does not match the required version: 2.0.0'
+      end
+
+      it 'Has no warning with semantic version match' do
+        stash_bundler_environment
+        ENV['BUNDLE_GEMFILE'] = 'mock argument'
+        ENV['BUNDLE_BIN_PATH'] = 'another mock argument'
+        ENV['BUNDLER_VERSION'] = '2.0.9'
+        Podfile.new do
+          ensure_bundler! '~> 2.0.0'
+        end
+        restore_bundler_environment
+        CoreUI.warnings.should.be.empty
+      end
+
+      it 'Skips version check with missing Bundle Version (edge case)' do
```

Please double-check this is how we want this case to play out (Version check silently skipped if `ENV['BUNDLER_VERSION']` is nil).

---

its fine its an edge case.